src/python/WMCore/Services/pycurl_manager.py

```diff
@@ -67,23 +67,43 @@ def __init__(self, config=None, logger=None):
         self.maxredirs = config.get('maxredirs', 5)
         self.logger = logger if logger else logging.getLogger()
 
+    def encode_params(self, params, verb, doseq):
+        """ Encode request parameters for usage with the 4 verbs.
+            Assume params is alrady encoded if it is a string and
+            uses a different encoding depending on the HTTP verb
+            (either json.dumps or urllib.urlencode)
+        """
+        #data is already encoded, just return it
+        if isinstance(params, basestring):
+            return params
+
+        #data is not encoded, we need to do that
+        if verb in ['GET', 'HEAD']:
+            if params:
+                encoded_data = urllib.urlencode(params, doseq=doseq)
+            else:
+                return ''
+        else:
+            if params:
+                encoded_data = json.dumps(params)
+            else:
+                return {}
+
+        return encoded_data
+
     def set_opts(self, curl, url, params, headers,
                  ckey=None, cert=None, capath=None, verbose=None, verb='GET', doseq=True, cainfo=None):
         """Set options for given curl object, params should be a dictionary"""
-        if  params == None:
-            params = {}
-        if  not isinstance(params, dict):
-            raise TypeError("pycurl parameters should be passed as dictionary")
+        if  not (isinstance(params, dict) or isinstance(params, basestring) or params==None):
```

Comment:
@mmascher how about updating this line to:
`if  not (isinstance(params, (dict, basestring)) or params is None):`
?